Sources/KlaviyoSwift/Klaviyo.swift

```diff
@@ -167,16 +168,18 @@ public class Klaviyo : NSObject {
     }
     
     /**
-     handlePush: Extracts tracking information from received push notification and sends the data to Klaviyo for push-tracking
-     analystics.
+     handlePush: Sends the push payload back to Klaviyo as an opened event
+        if the message originated from Klaviyo, as determined by presence of _k tracking parameters
      
      - Parameter userInfo: NSDictionary containing the push notification text & metadata
      */
     public func handlePush(userInfo: NSDictionary) {
-        if let metadata = userInfo["_k"] as? NSDictionary {
-            trackEvent(eventName: KLPersonOpenedPush, properties: metadata)
-        } else {
-            trackEvent(eventName: KLPersonOpenedPush, properties: userInfo as NSDictionary)
+        // Test it originated from klaviyo by checking presence of _k tracking parameters
+        if let body = userInfo["body"] as? NSDictionary, let _ = body["_k"] as? NSDictionary {
```

Comment:
❔ Curious why do we need to type cast  `body["_k"]` to a `NSDictionary` in order to see if body has the key `_k`?

Reply:
I suppose we don't, other than to make sure _k is a dict as we'd expect. When we circle back to this after noah's bigger changes, I can revisit